src/github.com/matrix-org/go-neb/services/rssbot/rssbot.go

```diff
@@ -302,7 +317,41 @@ func (s *Service) queryFeed(feedURL string) (*gofeed.Feed, []gofeed.Item, error)
 	return feed, items, nil
 }
 
+func containsAny(item string, list []string) bool {
+	for _, i := range list {
+		if strings.Contains(item, i) {
+			fmt.Println("contains", item, i)
+			return true
+		}
+	}
+	return false
+}
+
+func itemFiltered(i *gofeed.Item, mustInclude, mustNotInclude *includeRules) bool {
+	// If `must_include` terms have been specified, the feed item must include at least one item for each field
+	// that has been specified. This means that if both an Author and Title substring have been specified,
+	// both the Author and Title must contain these substrings or the item will be skipped.
+	if (i.Author != nil && len(mustInclude.Author) > 0 && !containsAny(i.Author.Name, mustInclude.Author)) ||
+		(len(mustInclude.Title) > 0 && !containsAny(i.Title, mustInclude.Title)) ||
+		(len(mustInclude.Description) > 0 && !containsAny(i.Description, mustInclude.Description)) {
+		return true
+	}
+
+	// If `must_not_include` terms have been specified, the feed item fields must not contain any of the
+	// substrings that were specified for each field.
+	if (i.Author != nil && containsAny(i.Author.Name, mustNotInclude.Author)) ||
+		containsAny(i.Title, mustNotInclude.Title) ||
+		containsAny(i.Description, mustNotInclude.Description) {
+		return true
+	}
+	return false
+}
+
 func (s *Service) newItems(feedURL string, allItems []*gofeed.Item) (items []gofeed.Item) {
+	mustInclude := s.Feeds[feedURL].MustInclude
+	mustNotInclude := s.Feeds[feedURL].MustNotInclude
+	fmt.Println(mustInclude, len(mustInclude.Description))
```

Comment:
Spurious logging

Reply:
Yup, I forgot a couple of these. Removed now.